pep0/pep.py

```diff
@@ -60,36 +60,48 @@ class Author(object):
             The author's email address.
     """
 
-    def __init__(self, author_and_email_tuple):
+    def __init__(self, author_and_email_tuple, authors_exceptions):
         """Parse the name and email address of an author."""
+        self.first = self.last = ''
+
         name, email = author_and_email_tuple
         self.first_last = name.strip()
         self.email = email.lower()
-        last_name_fragment, suffix = self._last_name(name)
-        name_sep = name.index(last_name_fragment)
-        self.first = name[:name_sep].rstrip()
-        self.last = last_name_fragment
-        if self.last[1] == u'.':
-            # Add an escape to avoid docutils turning `v.` into `22.`.
-            self.last = u'\\' + self.last
-        self.suffix = suffix
-        if not self.first:
-            self.last_first = self.last
+
+        name_dict = authors_exceptions.get(self.first_last)
+        if name_dict:
+            self.last_first = name_dict["Surname First"]
+            self.nick = name_dict["Name Reference"]
+        else:
+            self.set_name_parts()
+
+    def set_name_parts(self):
+        name_dict = self._parse_name(self.first_last)
+        suffix = name_dict.get("suffix")
+        if "name" in name_dict:
+            self.last_first = name_dict["name"]
+            self.nick = name_dict["name"]
         else:
-            self.last_first = u', '.join([self.last, self.first])
-            if self.suffix:
-                self.last_first += u', ' + self.suffix
+            self.first = name_dict["forename"].rstrip()
+            self.last = name_dict["surname"]
+            if self.last[1] == ".":
+                # Add an escape to avoid docutils turning `v.` into `22.`.
+                self.last = "\\" + self.last
+            self.last_first = ", ".join([self.last, self.first])
+            self.nick = self.last
+
+        if suffix:
+            self.last_first += f", {suffix}"
+
         if self.last == "van Rossum":
             # Special case for our beloved BDFL. :)
             if self.first == "Guido":
                 self.nick = "GvR"
             elif self.first == "Just":
                 self.nick = "JvR"
```

Comment:
I wonder if these special cases could be handled by the CSV file too.

Reply:
Done!